hydromt/components/grid.py

```diff
@@ -806,3 +777,35 @@ def test_equal(self, other: ModelComponent) -> tuple[bool, dict[str, str]]:
             errors["data"] = str(e)
 
         return len(errors) == 0, errors
+
+    def parse_region(
+        self,
+        region: dict,
+        *,
+        hydrography_fn: str,
+        basin_index_fn: str,
+        crs: Optional[int],
+    ) -> gpd.GeoDataFrame:
+        """Parse the region dict into a GeoDataFrame.
+
+        Overrides the base functionality to also support "grid".
+        Default options are still available from the base functionality.
+        """
+        kwargs = region.copy()
+        kind = next(iter(kwargs))  # first key of region
+        if kind == "grid":
+            value0 = kwargs.pop(kind)
+            ds = self._data_catalog.get_rasterdataset(value0, driver_kwargs=kwargs)
```

Comment:
Allowing for driver_kwargs in region dict is new and should be documented. Also driver_kwargs is used for the data catalog but I don't think this is the argument to use for get_rasterdataset. Unless v1 changed? I couldn't find the get_data method for rasterdataset...

Reply:
I think we should not introduce new functionalities at this point, so if this is new we should remove it again. Besides, any driver kwargs that are needed by default should be listed in the data catalog themselves if I remember correctly, so they are not really necessary at this point, though please correct me if I'm wrong.

Reply:
I have split the driver kwargs and the region_kwargs. Yaml will show:

```ymal
steps:
- grid.create:
    region_options:
      filename: region.geojson
      # other kwargs for the region driver
    # other kwargs for the grid driver
```